tools/kafka_ssl/generate_ssl_CA/utils/env_utils

```diff
@@ -0,0 +1,313 @@
+#!/bin/bash
+
+get_os(){
+  if [ -f "/etc/apt/sources.list" ]; then
+    if [ -f "/etc/linx-release" ]; then
+      os_version=$(cat /etc/linx-release | grep -o '[0-9]\.[0-9]\.[0-9]\{1,3\}' )
+      echo "ROCKY $os_version"
+    elif [ -f "/etc/lsb-release" ]; then
+      os_version=$(cat /etc/lsb-release | grep  "DISTRIB_RELEASE" | cut -d= -f2)
+      echo "UBUNTU $os_version"
+    elif [ -f "/etc/os-release" ]; then
+      os_version=$(cat /etc/os-release | grep  "VERSION_ID" | cut -d= -f2)
+      os_version=${os_version//\"}  # remove all double quotes
+      echo "DEBIAN $os_version"
+    fi
+  elif [ -d "/etc/yum.repos.d" ]; then
+    # Centos and RedHat are treated equally
+    if grep "Amazon Linux" /etc/system-release &>/dev/null; then
+      os_version=" 7.0"
+    else
+      variant="$(cat /etc/system-release | cut -d ' ' -f2)"
+      if [ "$variant" = "Stream" ]; then
+          os_version=" $(cat /etc/os-release | grep 'VERSION_ID=' | cut -d'"' -f 2)"
+      else
+          os_version="$(cat /etc/system-release | grep -o ' [0-9]\.[0-9]\{1,3\}')"
+      fi
+    fi
+    echo "RHEL$os_version"
+  elif [ -d "/etc/zypp/repos.d" ]; then
+    os_version=$(cat /etc/os-release | grep  "VERSION_ID" | cut -d= -f2)
+    os_version=${os_version//\"}  # remove all double quotes
+    echo "SUSE $os_version"
+  else
+    echo "UNKOWN OS"
+  fi
+}
+
+check_os(){
+  OS=$1
+  version=$2
+  note "OS obtained: $OS $version"
+  local error_msg="Unsupported OS. Current support CentOS 6.5 to 8.0; RedHat 6.5 to 9.0;"
+  if [ -z "$version" ]; then
+    error "Unknown OS version. $error_msg"
+    exit 1
+  fi
+
+  if [ "$OS" = "RHEL" ]; then
+   # the following one will end with one item array on docker centos 7.3, i.e. "${ver_arr[0]}" is "7 3"
+   # local ver_arr=(${version//./ })
+    IFS='.' read -r -a ver_arr <<< "$version"
+    if [[ "${ver_arr[0]}" -lt "6" || "${ver_arr[0]}" -eq "6" && "${ver_arr[1]}" -lt "5" ]]; then
+      error "$error_msg"
+      exit 1
+    else
+      note "OS check passed [OK]"
+    fi
+  else
+    error "$error_msg"
+    exit 1
+  fi
+}
+
+check_root(){
+  if [[ $EUID -ne 0 ]]; then
+    error "Sudo or root rights are required."
+    exit 1
+  fi
+}
+
+install_openJDK(){
+  if [[ "${SETUP_JDK}" == "true" ]]; then
+    if ! which java > /dev/null 2>&1; then
+      prog "start install openjdk-1.8.0."
+      yum install -y java-1.8.0-openjdk > /dev/null 2>&1
+    else
+    java_version=$(java -version 2>&1 | sed '1!d' | sed -e 's/"//g' | awk '{print $3}'| cut -d_ -f1)
```

Comment:
indent is wrong

Reply:
@dadongwang-tg 
I have modified it to support 1.8.0 and above openJDK version.
If it is lower than 1.8.0 and has the '- i' option, it will install openJDK_1.8.0.
If it is higher than 1.8.0 and has the '- i' option, skip the installation.